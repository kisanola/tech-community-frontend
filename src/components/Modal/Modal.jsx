import React from 'react';
import PropTypes from 'prop-types';
import './Modal.scss';

const Modal = ({ handleClose, show, children }) => {
  return (
    <div className="modal" style={{ display: show ? 'flex' : 'none' }}>
      <div className="dismiss" onClick={handleClose}>
        <span>&times;</span>
      </div>
      <div className="modal-dismiss" onClick={handleClose} />
      <section className="modal-main">{children}</section>
    </div>
  );
};

Modal.propTypes = {
  handleClose: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  children: PropTypes.node,
};

Modal.defaultProps = {
  children: null,
};

export default Modal;
