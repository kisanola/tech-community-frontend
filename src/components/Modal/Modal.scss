.modal {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);

  .modal-main {
    display: flex;
    justify-content: center;
    position: fixed;
    background: transparent;
    width: 50%;
    height: auto;
  }

  .dismiss {
    position: fixed;
    right: 1em;
    top: -0.2em;
    font-size: 3em;
    cursor: pointer;
    color: white;
    z-index: 5;
  }
  .modal-dismiss {
    position: absolute;
    left: 0;
    top: o;
    width: 100%;
    height: 100%;
  }
}
