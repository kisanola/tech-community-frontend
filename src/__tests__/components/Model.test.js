import React from 'react';
import { shallow, mount } from 'enzyme';
import Modal from '../../components/Modal/Modal';

let wrapper;
const props = {
  handleClose: jest.fn(),
  show: true,
};
describe('Modal.jsx', () => {
  beforeEach(() => {
    wrapper = mount(<Modal {...props} />);
  });
  test('should render Header.jx', () => {
    wrapper = shallow(<Modal {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
