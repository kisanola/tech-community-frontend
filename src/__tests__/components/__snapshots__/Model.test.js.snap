// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.jsx should render Header.jx 1`] = `ShallowWrapper {}`;
